controllers/quay/quayregistry_controller.go

```diff
@@ -120,10 +120,91 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
-		log.Info("migrations in progress, skipping reconcile")
-		return ctrl.Result{}, nil
+	created := v1.GetCondition(updatedQuay.Status.Conditions, v1.ConditionComponentsCreated)
+	migrationInProgress := created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress
+	if migrationInProgress {
+		log.Info("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+		log.Info("migration currently in progress!")
+		log.Info(fmt.Sprintf("currentVersion is: '%s'", updatedQuay.Status.CurrentVersion))
+
+		log.Info("checking Quay upgrade `Job` completion")
+
+		var upgradeJob batchv1.Job
+		err := r.Client.Get(
+			ctx,
+			types.NamespacedName{
+				Name:      updatedQuay.GetName() + "-" + v1.QuayUpgradeJobName,
+				Namespace: updatedQuay.GetNamespace(),
+			},
+			&upgradeJob,
+		)
+		if err != nil {
+			log.Error(err, "could not retrieve Quay upgrade `Job`")
+			return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+		}
+
+		if upgradeJob.Status.Active == 1 {
+			return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+		}
+
+		if upgradeJob.Status.Succeeded == 1 {
+			log.Info("Quay upgrade complete, updating `status.currentVersion`")
+
+			var freshQuay v1.QuayRegistry
+			if err := r.Client.Get(ctx, req.NamespacedName, &freshQuay); err != nil {
+				log.Error(err, "could not retrieve QuayRegistry")
+				return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+			}
+			qcopy := freshQuay.DeepCopy()
+
+			msg := "All registry components created"
+			condition := v1.Condition{
+				Type:               v1.ConditionComponentsCreated,
+				Status:             metav1.ConditionTrue,
+				Reason:             v1.ConditionReasonComponentsCreationSuccess,
+				Message:            msg,
+				LastUpdateTime:     metav1.Now(),
+				LastTransitionTime: metav1.Now(),
+			}
+			qcopy.Status.Conditions = v1.SetCondition(qcopy.Status.Conditions, condition)
+			qcopy.Status.CurrentVersion = v1.QuayVersionCurrent
+			r.EventRecorder.Event(qcopy, corev1.EventTypeNormal, string(v1.ConditionReasonHealthChecksPassing), msg)
+
+			if err = r.Client.Status().Update(ctx, qcopy); err != nil {
+				log.Error(err, "could not update QuayRegistry status with current version")
+				return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+			}
+
+			log.Info(fmt.Sprintf("currentVersion updated to: '%s'", qcopy.Status.CurrentVersion))
+
+			log.Info("successfully updated `status` after Quay upgrade")
+
+			return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+		}
+
+		// when the job fails, the next reconciliation to run will think
+		// that migrations are currently not running, since we change
+		// the condition reason to v1.ConditionReasonMigrationsFailed.
+		// this doesn't necessarily describe reality, because kube will
+		// retry failed jobs for us, but it's desired behaviour because
+		// when the migration job fails due to misconfiguration, then the
+		// reconcile function should be allowed to proceed.
```

Comment:
Adding to the comment - we're counting that the version check on [line 425](https://github.com/quay/quay-operator/blob/a94041d2b525518f5aa005e138eb7387a7019f94/controllers/quay/quayregistry_controller.go#L425-L431) will eventually be reached, setting the condition reason back to `v1.ConditionReasonMigrationsInProgress` and scheduling another reconcile. After the condition is set back to migration in progress we'll check the upgrade job again.

Reply:
Loved the comment!